biggerquery/job.py

```diff
@@ -0,0 +1,200 @@
+from __future__ import absolute_import
+
+
+import os
+import runpy
+import mock
+
+import tempfile
+import imp
+from types import ModuleType
+from inspect import getargspec
+from inspect import getmodule
+from unittest import TestCase
+from unittest import main
+
+from .utils import secure_create_dataflow_manager_import
+from .dataset_manager import create_dataset_manager
+create_dataflow_manager = secure_create_dataflow_manager_import()
+
+from .utils import unzip_file_and_save_outside_zip_as_tmp_file
+from .utils import AutoDeletedTmpFile
+from .configuration import DatasetConfig
+from .configuration import DataflowConfig
+
+DEFAULT_RETRY_COUNT = 3
+DEFAULT_RETRY_PAUSE_SEC = 60
+
+
+class Job(object):
+    def __init__(self,
+                 component,
+                 id=None,
+                 retry_count=DEFAULT_RETRY_COUNT,
+                 retry_pause_sec=DEFAULT_RETRY_PAUSE_SEC,
+                 **dependency_configuration):
+        self.id = id or component.__name__
+        self.component = component
+        self.dependency_configuration = dependency_configuration
+        self.retry_count = retry_count
+        self.retry_pause_sec = retry_pause_sec
+
+    def __call__(self, runtime):
+        return self.run(runtime)
+
+    def run(self, runtime):
+        return self._run_component(self._build_dependencies(runtime))
+
+    def _build_dependencies(self, runtime):
+        return {
+            dependency_name: self._build_dependency(
+                dependency_config=self._find_config(dependency_name),
+                runtime=runtime)
+            for dependency_name in self._component_dependencies
+        }
+
+    def _run_component(self, dependencies):
+        if not self._is_dataflow_job:
+            return self.component(**dependencies)
+        else:
+            return self._run_beam_component(dependencies)
+
+    def _run_beam_component(self, dependencies):
+        component_file_path = os.path.abspath(getmodule(self.component).__file__)
+        return runpy.run_path(
+            path_name=unzip_file_and_save_outside_zip_as_tmp_file(component_file_path, suffix='.py').name
```

Comment:
I think it would be better to create separate class for ziping/unziping files in the whole project.

Reply:
why?

Reply:
We've got methods for ziping/unziping in `utils.py` and in `deployments.py`. It will be more readable if we put all this code in one place

Reply:
So I will move zip stuff from the deployment.py to utils.

Reply:
done